starknet_py/net/udc_deployer/deployer.py

```diff
@@ -0,0 +1,146 @@
+# pyright: reportGeneralTypeIssues=false
+
+from typing import Optional, List, Union
+
+from starkware.starknet.definitions.fields import ContractAddressSalt
+from starkware.starknet.public.abi import get_selector_from_name
+
+from starknet_py.net import AccountClient
+from starknet_py.net.client_models import Hash, InvokeFunction, Call, Event
+from starknet_py.net.models import AddressRepresentation
+from starknet_py.net.models.deployer_addresses import deployer_address_from_network
+from starknet_py.utils.contructor_args_translator import translate_constructor_args
+from starknet_py.utils.data_transformer.universal_deployer_serializer import (
+    universal_deployer_serializer,
+    deploy_contract_abi,
+    deploy_contract_event_abi,
+)
+
+
+class Deployer:
+    """
+    Deployer used to deploy contracts through Universal Deployer Contract (UDC)
+    """
+
+    def __init__(
+        self,
+        account: AccountClient,
+        address: Optional[AddressRepresentation] = None,
+        salt: Optional[int] = None,
+        unique: bool = True,
+    ):
+        """
+        :param account: AccountClient used to sign and send transactions
+        :param address: Address of the UDC. Must be set when using a custom network
+        :param salt: The salt for a contract to be deployed. Random value is selected if it is not provided
+        :param unique: Boolean determining if the salt should be connected with the account's address. Default to True
+        """
+        address = (
+            address if isinstance(address, int) or address is None else int(address, 16)
+        )
+        address = address or deployer_address_from_network(
+            net=account.net, deployer_address=address
+        )
```

Comment:
What if address is None and unique is set to False? Will an exception be raised?

Reply:
Unique is not connected with the address so yes, an exception will be raised

Reply:
Right, an exception would be raised only if the address would be `None` and the account.net would be a custom network